src/structures/channel.ts

```diff
@@ -151,6 +151,11 @@ export class GuildChannel extends Channel {
     this.parentID = data.parent_id ?? this.parentID
   }
 
+  delete(): this {
+    this.client.rest.endpoints.deleteChannel(this.id)
```

Comment:
Delete needs to be async and you need to await deleteChannel.

Reply:
oh yeah, forget